.gitignore

```diff
@@ -0,0 +1,387 @@
+nx-X11/programs/Xserver/GL/mesa/swrast_setup/ss_triangle.c
+nx-X11/programs/Xserver/GL/mesa/swrast_setup/ss_triangle.h
+nx-X11/programs/Xserver/GL/mesa/swrast_setup/ss_tritmp.h
+nx-X11/programs/Xserver/GL/mesa/swrast_setup/swrast_setup.h
+nx-X11/programs/Xserver/GL/mesa/tnl/DONE
+nx-X11/programs/Xserver/GL/mesa/tnl/t_array_api.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_array_api.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_array_import.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_array_import.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_context.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_context.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_pipeline.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_pipeline.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_save_api.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_save_api.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_save_loopback.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_save_playback.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_arbprogram.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_arbprogram.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_arbprogram_sse.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_cliptmp.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_cull.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_fog.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_light.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_lighttmp.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_normals.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_points.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_program.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_render.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_rendertmp.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_texgen.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_texmat.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vb_vertex.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vertex.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vertex.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vertex_generic.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vertex_sse.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vp_build.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vp_build.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vtx_api.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vtx_api.h
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vtx_eval.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vtx_exec.c
+nx-X11/programs/Xserver/GL/mesa/tnl/t_vtx_generic.c
+nx-X11/programs/Xserver/GL/mesa/tnl/tnl.h
+nx-X11/programs/Xserver/hw/nxagent/a.out
+nx-X11/programs/Xserver/hw/nxagent/file1.h
+nx-X11/programs/Xserver/hw/nxagent/file2.h
+nx-X11/programs/Xserver/hw/nxagent/file3.c
+nx-X11/programs/Xserver/hw/nxagent/miinitext.c
+nx-X11/programs/Xserver/hw/nxagent/stubs.c
+nx-X11/programs/Xserver/hw/nxagent/xpstubs.c
+nx-X11/programs/Xserver/include/compiler.h
+nx-X11/programs/Xserver/include/osdep.h
+nx-X11/programs/Xserver/include/xf86Version.h
+nx-X11/programs/Xserver/include/xf86_ansic.h
+nx-X11/programs/Xserver/include/xf86_libc.h
+nx-X11/programs/Xserver/include/xkbfile.h
+nx-X11/programs/Xserver/xkb/modinit.h
```

Comment:
@sunweaver's point is that after running `clean/distclean`, all stray files should be gone and running `git status` is a way to list anything untracked. If this doesn't come up empty, we've forgotten some stuff.

While that might be true, I'm more often than not also using `git status` in an unclean state and having a `.gitignore` file certainly helps with the huge output...

Btw, did you know that every (sub-)directory can have its own `.gitignore` file? Instead of putting everything into the "root" file, I think I'll split it up.